### PR TITLE
[New Rule] Administrator Role Assigned to Okta User

### LINKED ISSUE #397: [New Rule] Administrator Role Assigned to Okta User
## Description

An adversary may attempt to assign an administrator role to an Okta user in order to assign additional permissions to a user account and maintain a presence in their target's environment.

```
event.dataset:okta.system and event.action:user.account.privilege.grant
```

## Required Info

- **Eventing Sources:**

`filebeat-*`, `logs-okta*`

- **Platforms**

`Okta`

- **Target ECS Version:** `1.6.0`
- **New fields required in ECS for this?** No
- **Related issues or PRs**

## Optional Info
- **References:**
- https://developer.okta.com/docs/reference/api/system-log/"
- https://developer.okta.com/docs/reference/api/event-types/

### PR DESCRIPTION
**Pull request by threat-punter**
## Issues

Resolves #397 

## Summary
Identifies when an administrator role is assigned to an Okta user. An adversary may attempt to assign an administrator role to an Okta user in order to assign additional permissions to a user account and maintain access to their target's environment.

## Contributor checklist

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/detection-rules/blob/main/CONTRIBUTING.md)?

### CODE REVIEW

**Comment by threat-punter** [comment]
@brokensound77 @bm11100 shall I open another issue to tune our existing rules to include similar information in the notes field? `The Okta Fleet integration or Filebeat module must be enabled to use this rule.`

**Comment by bm11100** [comment]
> @brokensound77 @bm11100 shall I open another issue to tune our existing rules to include similar information in the notes field? `The Okta Fleet integration or Filebeat module must be enabled to use this rule.`

I think that's a great idea. We added the index, but never modified the notes field.

File: rules/okta/persistence_administrator_role_assigned_to_okta_user.toml [hunk 1]
```diff
@@ -0,0 +1,53 @@
+[metadata]
+creation_date = "2020/11/06"
+ecs_version = ["1.6.0"]
+maturity = "development"
```
Comment (bm11100) [inline rules/okta/persistence_administrator_role_assigned_to_okta_user.toml:4]:
```suggestion
maturity = "production"
```

Comment (threat-punter) [reply rules/okta/persistence_administrator_role_assigned_to_okta_user.toml:4]: doh

File: rules/okta/persistence_administrator_role_assigned_to_okta_user.toml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+]
+index = ["filebeat-*", "logs-okta*"]
+language = "kuery"
+license = "Elastic License"
+name = "Administrator Role Assigned to Okta User"
+note = "The Okta Fleet integration or Filebeat module must be enabled to use this rule."
+references = [
+    "https://developer.okta.com/docs/reference/api/system-log/",
+    "https://developer.okta.com/docs/reference/api/event-types/",
+    "https://help.okta.com/en/prod/Content/Topics/Security/administrators-admin-comparison.htm",
```
Comment (threat-punter) [inline rules/okta/persistence_administrator_role_assigned_to_okta_user.toml:28]:
```suggestion
    "https://help.okta.com/en/prod/Content/Topics/Security/administrators-admin-comparison.htm",
    "https://developer.okta.com/docs/reference/api/system-log/",
    "https://developer.okta.com/docs/reference/api/event-types/",
```

File: rules/okta/persistence_administrator_role_assigned_to_okta_user.toml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+false_positives = [
+    """
+    Administrator roles may be assigned to Okta users by a Super Admin user. Verify that the behavior was expected.
+    Exceptions can be added to this rule to filter expected behavior.
+    """,
+]
+index = ["filebeat-*", "logs-okta*"]
+language = "kuery"
+license = "Elastic License"
+name = "Administrator Role Assigned to Okta User"
```
Comment (threat-punter) [inline rules/okta/persistence_administrator_role_assigned_to_okta_user.toml:23]:
```suggestion
name = "Administrator Role Assigned to an Okta User"
```

**Review by bm11100 — APPROVED** [review]
🆗

**Review by Samirbous — APPROVED** [review]

**Pull request merged**